### ISSUE TITLE
Set unitSystem conversion

### ISSUE THREAD

**Issue by pvandriessche** [josdejong/mathjs#1282]
I'm trying to set a unitSystem for a Unit that was created in "Inch", when I call setUnitSystem('si') the unit system is corretly changed. 
However when calling the format() the unit is still represented in Inch.
Is there a way to solve this?

This is the code
`    var measure = math.unit("2 in")
    console.log(measure.formatUnits());
//"in"
    measure.constructor.setUnitSystem("si");
    console.log("unit system :" + measure.constructor.getUnitSystem());
// "unit system: si"
    console.log(measure.formatUnits());`
// "in" => should be "m"?
Thanks in advance!

**Comment by josdejong** [comment]
@ericman314 you assigned yourself to this issue, did you have a look at it?

**Comment by ericman314** [comment]
Sorry, I forgot about this. Perhaps before trying a fix for this, we should look more closely at #1300 and find a good solution there that will address this and other related issues.

**Comment by josdejong** [comment]
Yes I think so. There are also some other ideas for improvements of units floating around here and there (#1305, #1306, #1041, maybe others). Can be helpful to keep these in mind too.
